### ISSUE TITLE
Hue controls broken in Tangent

### ISSUE THREAD

**Issue by latenitefilms** [fcpxhacks/fcpxhacks#1624]
‎Claus Andersen‎ [writes](
https://www.facebook.com/groups/commandpost/permalink/2198210423767990/):

> Yeeees! Wave 2 is here 🎉😍 Seems like all controls are working (except HUE). Awesome work Chris Hocking 🙏🙏
> The only quirk: Color Hue does'nt seem to work. When I turn the knob, it jumps to a random low value and stops. Really strange. But oh well, not often that I use the Hue adjustments.

**Comment by latenitefilms** [comment]
I can't reproduce this - Colour Wheels Hue Control is working fine on the Element.

@randomeizer - Wanna see if you can reproduce on the Wave?

**Comment by latenitefilms** [comment]
FYI: Trying to fault find on Facebook [here](https://www.facebook.com/groups/commandpost/permalink/2233952116860487/?comment_id=2234288603493505&reply_comment_id=2234777616777937&comment_tracking=%7B%22tn%22%3A%22R2%22%7D).

**Comment by latenitefilms** [comment]
Chatting with Claus on Facebook, the problem seems to be that `require("cp.apple.finalcutpro"):inspector():color():colorWheels():hue()` returns `nil`, but I'm not sure why as:

```lua
> hs.inspect(require("cp.apple.finalcutpro"):inspector():color():colorWheels():hueRow()._children)
2019-03-14 11:32:47: -- Loading extension: inspect
{ <userdata 1> -- hs._asm.axuielement: AXStaticText (0x60001599a488), <userdata 2> -- hs._asm.axuielement: AXSlider (0x60001599a518), <userdata 3> -- hs._asm.axuielement: AXTextField (0x60001599a5d8), <userdata 4> -- hs._asm.axuielement: AXStaticText (0x60001599a4b8), <userdata 5> -- hs._asm.axuielement: AXButton (0x60001599a4e8) }

> require("cp.apple.finalcutpro"):inspector():color():colorWheels():hueTextField():value()
nil
```

It's also interesting that in the below video the values are shown as "0,1" as opposed to "0.1" - which must be a region thing.

Video Demonstration on [Facebook](
https://www.facebook.com/groups/commandpost/permalink/2233952116860487/?comment_id=2234288603493505&reply_comment_id=2234782150110817&comment_tracking=%7B%22tn%22%3A%22R1%22%7D).

**Error Log:**
```lua
CommandPost v1.0.0-beta.55

Date Built:                     Mar 13 2019, 13:42:41
macOS Version:                  10.14.3
CommandPost Locale:             en
Current Keyboard Layout:        Danish
Final Cut Pro Path:             /Applications/Final Cut Pro.app
Final Cut Pro Version:          10.4.5
Final Cut Pro Language:         en
Developer Mode:                 false

2019-03-13 20:02:47: 2019-03-13 20:02:46: -- Loading /Users/claus/Applications/CommandPost.app/Contents/Resources/extensions/cp/init.lua
2019-03-13 20:02:46: -- Loading extension: uielement
2019-03-13 20:02:47: 20:02:47         cp:     Loading Plugins...
2019-03-13 20:02:50: 20:02:50 ** Warning:   plugins: Slow Plugin: finalcutpro.setup.scan (0.703615)
2019-03-13 20:02:51: 20:02:51 ** Warning:   plugins: Slow Plugin: core.language (0.52932)
2019-03-13 20:02:51:                  cp:     Plugins Loaded.
2019-03-13 20:02:51: -- Done.
2019-03-13 20:02:53: 20:02:53 tangentMan:     Connection To Tangent Hub (127.0.0.1:64246) successfully established.
2019-03-13 20:02:53:                          InitiateComms Received:
2019-03-13 20:02:53:                              Protocol Revision: 6
2019-03-13 20:02:53:                              Number of Panels: 6
2019-03-13 20:02:53:                                  Panel Type: Ripple (0x00110001)
2019-03-13 20:02:53:                                  Panel Type: Element-Tk (0x000c0001)
2019-03-13 20:02:53:                                  Panel Type: Element-Mf (0x000d0001)
2019-03-13 20:02:53:                                  Panel Type: Element-Kb (0x000e0001)
2019-03-13 20:02:53:                                  Panel Type: Element-Bt (0x000f0001)
2019-03-13 20:02:53:                                  Panel Type: Wave (0x000a0001)

> require("cp.apple.finalcutpro"):inspector():color():colorWheels():hue()
nil
```

**Comment by latenitefilms** [comment]
Interestingly...

```lua
require("cp.apple.finalcutpro"):inspector():color():colorWheels():hueRow()._children[3]
hs._asm.axuielement: *AXRole undefined* (0x60001599a5d8)
```

![aaaa](https://user-images.githubusercontent.com/22286696/54351173-017b9280-46a3-11e9-86cc-b85aa23913b6.jpg)

**Comment by latenitefilms** [comment]
Even more strangely:

```lua
> print(hs.inspect(require("cp.apple.finalcutpro"):inspector():color():colorWheels():hueRow()._children)); print(require("cp.apple.finalcutpro"):inspector():color():colorWheels():hueRow()._children[3]); print(require("cp.apple.finalcutpro"):inspector():color():colorWheels():hueRow()._children[3]:value())
2019-03-14 11:51:00: { <userdata 1> -- hs._asm.axuielement: *AXRole undefined* (0x60001599a488), <userdata 2> -- hs._asm.axuielement: *AXRole undefined* (0x60001599a518), <userdata 3> -- hs._asm.axuielement: *AXRole undefined* (0x60001599a5d8), <userdata 4> -- hs._asm.axuielement: *AXRole undefined* (0x60001599a4b8), <userdata 5> -- hs._asm.axuielement: *AXRole undefined* (0x60001599a4e8) }
2019-03-14 11:51:00: hs._asm.axuielement: *AXRole undefined* (0x60001599a5d8)
[string "print(hs.inspect(require("cp.apple.finalcutpr..."]:1: attempt to call a nil value (method 'value')
stack traceback:
[string "print(hs.inspect(require("cp.apple.finalcutpr..."]:1: in main chunk
[C]: in function 'xpcall'
...app/Contents/Resources/extensions/hs/_coresetup/init.lua:523: in function <...app/Contents/Resources/extensions/hs/_coresetup/init.lua:499>
```

**Comment by latenitefilms** [comment]
```lua
> local contentUI = require("cp.apple.finalcutpro"):inspector():color():colorWheels():contentUI()
print(string.format("contentUI: %s", contentUI))
print(string.format("children: %s", contentUI and hs.inspect(contentUI:children())))
2019-03-14 12:47:31: contentUI: hs._asm.axuielement: AXScrollArea (0x60003c0dec38)
2019-03-14 12:47:31: children: { <userdata 1> -- hs._asm.axuielement: AXMenuButton (0x60003c0df508), <userdata 2> -- hs._asm.axuielement: AXGroup (0x60003c0df538), <userdata 3> -- hs._asm.axuielement: AXGroup (0x60003c0df568), <userdata 4> -- hs._asm.axuielement: AXGroup (0x60003c0df598), <userdata 5> -- hs._asm.axuielement: AXGroup (0x60003c0df5c8), <userdata 6> -- hs._asm.axuielement: AXStaticText (0x60003c0df5f8), <userdata 7> -- hs._asm.axuielement: AXSlider (0x60003c0df628), <userdata 8> -- hs._asm.axuielement: AXTextField (0x60003c0df658), <userdata 9> -- hs._asm.axuielement: AXStaticText (0x60003c0df688), <userdata 10> -- hs._asm.axuielement: AXButton (0x60003c0df6b8), <userdata 11> -- hs._asm.axuielement: AXStaticText (0x60003c0df6e8), <userdata 12> -- hs._asm.axuielement: AXSlider (0x60003c0df718), <userdata 13> -- hs._asm.axuielement: AXTextField (0x60003c0df748), <userdata 14> -- hs._asm.axuielement: AXStaticText (0x60003c0df778), <userdata 15> -- hs._asm.axuielement: AXButton (0x60003c0df7a8), <userdata 16> -- hs._asm.axuielement: AXStaticText (0x60003c0df7d8), <userdata 17> -- hs._asm.axuielement: AXStaticText (0x60003c0df808), <userdata 18> -- hs._asm.axuielement: AXButton (0x60003c0df838), <userdata 19> -- hs._asm.axuielement: AXSlider (0x60003c0df868), <userdata 20> -- hs._asm.axuielement: AXTextField (0x60003c0df898), <userdata 21> -- hs._asm.axuielement: AXStaticText (0x60003c0df8c8), <userdata 22> -- hs._asm.axuielement: AXButton (0x60003c0df8f8), <userdata 23> -- hs._asm.axuielement: AXCheckBox (0x60003c0df928), <userdata 24> -- hs._asm.axuielement: AXStaticText (0x60003c0df958), <userdata 25> -- hs._asm.axuielement: AXButton (0x60003c0df988), <userdata 26> -- hs._asm.axuielement: AXCheckBox (0x60003c0df9b8), <userdata 27> -- hs._asm.axuielement: AXStaticText (0x60003c0df9e8), <userdata 28> -- hs._asm.axuielement: AXButton (0x60003c0dfa18), <userdata 29> -- hs._asm.axuielement: AXCheckBox (0x60003c0dfa48), <userdata 30> -- hs._asm.axuielement: AXStaticText (0x60003c0dfa78), <userdata 31> -- hs._asm.axuielement: AXButton (0x60003c0dfaa8), <userdata 32> -- hs._asm.axuielement: AXCheckBox (0x60003c0dfad8), <userdata 33> -- hs._asm.axuielement: AXStaticText (0x60003c0dfb08), <userdata 34> -- hs._asm.axuielement: AXSlider (0x60003c0dfb38), <userdata 35> -- hs._asm.axuielement: AXTextField (0x60003c0dfb68), <userdata 36> -- hs._asm.axuielement: AXStaticText (0x60003c0dfb98), <userdata 37> -- hs._asm.axuielement: AXButton (0x60003c0dfbc8), <userdata 38> -- hs._asm.axuielement: AXScrollBar (0x60003c0dfbf8) }
```

**Comment by latenitefilms** [comment]
Will be solved in #1747
